# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/sdn/sdnAccessible.cpp
# Commit: 1a948c0ec982
# Full Hash: 1a948c0ec9826d73bba860699f540cbb25f34e34
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-04-29 21:42:31
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 18: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 3ea7b46bb4f8 -r 1a948c0ec982 accessible/windows/sdn/sdnAccessible.cpp
--- a/accessible/windows/sdn/sdnAccessible.cpp	Thu Apr 29 10:20:56 2021 +0000
+++ b/accessible/windows/sdn/sdnAccessible.cpp	Thu Apr 29 10:20:57 2021 +0000
@@ -52,7 +52,7 @@
     return S_OK;
   }
 
-  AccessibleWrap* accessible = GetAccessible();
+  MsaaAccessible* accessible = GetMsaa();
   if (accessible) return accessible->QueryInterface(aREFIID, aInstancePtr);
 
   // IUnknown* is the canonical one if and only if this accessible doesn't have
@@ -105,9 +105,9 @@
   // application can compare this to the childID we return for events such as
   // focus events, to correlate back to data nodes in their internal object
   // model.
-  AccessibleWrap* accessible = GetAccessible();
+  MsaaAccessible* accessible = GetMsaa();
   if (accessible) {
-    *aUniqueID = MsaaAccessible::GetChildIDFor(accessible);
+    *aUniqueID = MsaaAccessible::GetChildIDFor(accessible->LocalAcc());
   } else {
     if (mUniqueId.isNothing()) {
       MsaaAccessible::AssignChildIDTo(WrapNotNull(this));